hal/src/rtl872x/usart_hal.cpp

```diff
@@ -743,8 +743,8 @@ class Usart {
                 GDMA_BASE->CH[rxDmaInitStruct_.GDMA_ChNum].CFG_LOW |= BIT_CFGX_LO_CH_SUSP;
                 __DSB();
                 __ISB();
-                while (GDMA_GetDstAddr(rxDmaInitStruct_.GDMA_Index, rxDmaInitStruct_.GDMA_ChNum) < expectedDstAddr) {
-                    // XXX: spin around, this should be pretty fast
+                while (GDMA_GetDstAddr(rxDmaInitStruct_.GDMA_Index, rxDmaInitStruct_.GDMA_ChNum) < expectedDstAddr && 
+                      (GDMA_BASE->ChEnReg & (1 << rxDmaInitStruct_.GDMA_ChNum))) {
```

Comment:
Won't it lose bytes? for example, if we re goinng to consum 3 more bytes according to `RX_BYTE_CNT`, but here we just ignore that if the channel is closed, will the 3 bytes get missed?

Reply:
The assumption is that the bytes have already been transferred.
We enter this loop with knowing that the UART reported `RX_BYTE_CNT` should be processed by the DMA controller. For unknown reasons, sometimes `GDMA_GetDstAddr()` does not increment by this count. In those cases we look at this bit. 
According to the datasheet, this bit is cleared by the DMA controller when the transfer is complete. So if this is complete we assume that the data must be present, regardless of if `GDMA_GetDstAddr()` indicates the correct address:
![Screen Shot 2022-12-15 at 1 12 41 PM](https://user-images.githubusercontent.com/86268380/207968536-c867b225-869e-4381-ab17-9f687e986c21.png)

Reply:
Let's add some comments then, in case of forgetting why we're doing in this way over time.

Reply:
Let's definitely add some comments here.